File: scripts/archive/count_framing.mjs
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
import path from 'path';

// Load from tracenews-api/.env
dotenv.config({ path: '/Users/emekaabraham/Downloads/tracenews-api/.env' });

const url = process.env.SUPABASE_URL;
const key = process.env.SUPABASE_SERVICE_KEY || process.env.SUPABASE_KEY;

if (!url || !key) {
  console.error("Missing credentials");
  process.exit(1);
}

const supabase = createClient(url, key);

async function run() {
  const { count, error } = await supabase
    .from('clusters')
    .select('id', { count: 'exact', head: true })
    .gte('created_at', '2026-07-09T00:00:00Z')
    .not('framing_cache', 'is', null);
    
  if (error) {
    console.error(error);
  } else {
    console.log(`Count of framed clusters since July 9: ${count}`);
  }
}
run();

File: scripts/archive/count_framing.py
import os
from supabase import create_client

url = os.environ.get("SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY") or os.environ.get("SUPABASE_KEY")

if not url or not key:
    print("No Supabase credentials found in env.")
    # let's try to load from tracenews-api/.env
    from dotenv import load_dotenv
    load_dotenv("/Users/emekaabraham/Downloads/tracenews-api/.env")
    url = os.environ.get("SUPABASE_URL")
    key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY") or os.environ.get("SUPABASE_KEY")

if not url or not key:
    print("Still no credentials.")
    exit(1)

supabase = create_client(url, key)

res = supabase.table("clusters").select("id", count="exact").gte("created_at", "2026-07-09T00:00:00Z").not_.is_("framing_cache", "null").execute()
print(f"Count of framed clusters since July 9: {res.count}")
